user/user.go

```diff
@@ -2,19 +2,39 @@ package main
 
 import (
 	"encoding/json"
+	"fmt"
 	"log"
 	"net/http"
+	"strconv"
 
+	"github.com/TempleEight/spec-golang/user/dao"
 	"github.com/gorilla/mux"
 )
 
 func main() {
 	r := mux.NewRouter()
-	r.HandleFunc("/user", userGetHandler).Methods(http.MethodGet)
+	r.HandleFunc("/user/{id}", userGetHandler).Methods(http.MethodGet)
 
 	log.Fatal(http.ListenAndServe(":80", r))
 }
 
 func userGetHandler(w http.ResponseWriter, r *http.Request) {
-	json.NewEncoder(w).Encode("Hello, World")
+	userIDStr := mux.Vars(r)["id"]
+	if len(userIDStr) == 0 {
+		http.Error(w, "No user ID provided", http.StatusBadRequest)
+		return
+	}
+
+	userID, err := strconv.ParseInt(userIDStr, 10, 64)
+	if err != nil {
+		http.Error(w, "Invalid user ID provided", http.StatusBadRequest)
+		return
+	}
+
+	user, err := dao.GetUser(userID)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Something went wrong: %s", err.Error()), http.StatusInternalServerError)
```

Comment:
User doesn’t exist sounds like a 4xx (404?), not a 500

Reply:
Yeah good point, I guess there's actually multiple errors that could happen here, so might be worth defining those types and switching on them